### PR TITLE
Add testing for PyTorch string I/O

### PR DESCRIPTION
**Pull request by krishung5**

### CODE REVIEW

File: qa/common/test_util.py [hunk 1]
```diff
@@ -150,15 +150,17 @@ def validate_for_libtorch_model(input_dtype,
...
+    # batching when 'reshape' field is set properly to empty shape.
     if ((input_dtype == np.object_) or (output0_dtype == np.object_) or
-        (output1_dtype == np.object_)) and ((len(input_shape) > 1) or
-                                            (len(output0_shape) > 1) or
-                                            (len(output1_shape) > 1) or
-                                            (max_batch != 0)):
+        (output1_dtype == np.object_)) and (((len(input_shape) > 1) or
+                                             (len(output0_shape) > 1) or
+                                             (len(output1_shape) > 1) or
+                                             (max_batch != 0)) and not reshape):
```
Comment (GuanLuo) [inline qa/common/test_util.py:163]: Can you break this down to separate condition for readability? i.e.
```
has_string_type = ....
is_one_dimensional = ...
if has_string_type and .... :
    return False
```

File: qa/L0_infer_reshape/infer_reshape_test.py [hunk 2]
```diff
@@ -135,15 +135,19 @@ def _full_reshape(self,
...
+                                          input_shapes[0],
+                                          input_shapes[0],
+                                          reshape=True):
             # skip variable size reshape on libtorch for now,
             # see "gen_qa_reshape_model.py" for detail
             if dtype != np.int32:
                 # model that does not support batching
-                if no_batch:
+                # skip for libtorch string I/O
+                if (no_batch) and (dtype != np_dtype_string):
```
Comment (CoderHam) [inline qa/L0_infer_reshape/infer_reshape_test.py:150]: do not use brackets for a single arg

**Review by CoderHam — CHANGES_REQUESTED** [review]
You will also need to add the updated (reshaped string i/o libtorch) models to dlcluster

File: qa/common/test_util.py (hunk 1 above)
Comment (krishung5) [reply qa/common/test_util.py]: Done

File: qa/L0_infer_reshape/infer_reshape_test.py (hunk 2 above)
Comment (krishung5) [reply qa/L0_infer_reshape/infer_reshape_test.py]: Removed brackets

**Comment by krishung5** [comment]
Updated the string I/O model to dlcluster. CI passes, please review. Will revert the temporarily testing repo dir back.

File: qa/L0_infer_reshape/test.sh [hunk 3]
```diff
@@ -38,6 +38,7 @@ if [ ! -z "$TEST_REPO_ARCH" ]; then
     REPO_VERSION=${REPO_VERSION}_${TEST_REPO_ARCH}
 fi
 
+REPO_VERSION="22.07dev_krish"
```
Comment (GuanLuo) [inline qa/L0_infer_reshape/test.sh:41]: Drop this when ready to merge

**Review by GuanLuo — APPROVED** [review]

**Review by tanmayv25 — DISMISSED** [review]

**Review by CoderHam — DISMISSED** [review]

**Comment by krishung5** [comment]
Dropped the testing repo tag. Please re-approve, thanks

**Pull request merged**